### DISCUSSION TITLE
Make the node executable an external resource

### DISCUSSION THREAD

**Discussion by leonardospeca** [yao-pkg/pkg#98] [Ideas]
Hi. I use pkg in every project and I like it a lot. It has only one little problem for me:
Even if the application is really small, when I make a change, I need to upload over FTP 80mb of executable, where most of the size is the node executable. I use pkg to "hide" the source code of nodejs, so in your opinion, is possible to have the Exe with the application, that launch node from a path of the operating system?

**Comment by robertsLando** [comment]
In theory you could use the same technique pkg uses to inject the application bytecode into the nodejs exe to inject the updated code into the existing app but that would be so complicated and IMO the best alternative for you is to enable compression, that will save you lot space

**Reply by leonardospeca** [reply]
Yes it seems complicated to inject only the updates..

So is not possible to us Pkg only to create the "hidden file system" for the application, and then call nodejs executable from an external path? Could be a nice idea?

Electron JS have the executable in the main folder for example, then the file system of the application is inside /resources/app.asar file, so an update of the application consist on replace the app.asar file (that is a lot smaller then the entire application)

**Reply by robertsLando** [reply]
@leonardospeca It doesn't support this and I have no time to implement such thing
